apisix/admin/plugins.lua

```diff
@@ -42,7 +44,15 @@ end
 
 function _M.get(name)
     local arg = get_uri_args()
-    if arg and arg["all"] == "true" then
+    -- If subsystem is passed inside args then it should be oneOf: http / stream.
+    local subsystem = arg["subsystem"]
+    if subsystem and subsystem ~= "http" and subsystem ~= "stream" then
+        return 400, {error_msg = "unsupported subsystem: "..subsystem}
+    end
+
+    -- arg all to be deprecated
+    if (arg and arg["all"] == "true") or not name then
```

Comment:
```suggestion
    if (arg and arg["all"] == "true") then
```
There is no need to change the original condition

Reply:
@monkeyDluffy6017 This condition is so that `/plugins` can also return what `/plugins?all=true` returns currently. Adding this or condition is important to make the behaviour of `/plugins` == behaviour of `plugins?all=true`

Reply:
`not name` is basically like passing `/apisix/admin/plugins`

Reply:
Applied the suggestion